# 天池-新人实战赛之[离线赛]


In [3]:
import pandas as pd, numpy as np
import matplotlib.pyplot as plt
#Jupyter 设置
%matplotlib inline

In [42]:
#读取输入
df_user_data = pd.read_csv('data/tianchi_fresh_comp_train_user.csv')
df_item_data = pd.read_csv('data/tianchi_fresh_comp_train_item.csv')

# df_user_data = pd.read_csv('data/test_user.csv',names=['user_id','item_id','behavior_type','user_geohash','item_category','time'])
print(len(df_user_data))
# df_user_data

23291027


In [43]:
dict={}
for i in range(len(df_user_data)):
   
    if(i <1):
        continue
    userid = df_user_data['user_id'][i]
    ls = dict.get(userid,0)
    if(ls):
        dict[userid]=userid
    else:
        dict[userid]=userid
        
print(len(dict))


20000


In [49]:
i=0
for (k,v) in  dict.items(): 
    if(i==18000):
        print ("dict[%s]=" % k,v )
    i=i+1

dict[37047140]= 37047140


统计数据，总共有23291027数据，2w个用户，将1.8w的用户作为训练数据集，0.2W条数据作为验证数据集,根据计算应该是以user_id =37047140 为分割


behavior_type 	
 包括浏览、收藏、加购物车、购买，对应取值分别是1、2、3、4。
 
 一数据处理
 获取数据，根据userid 分组 时间排序排序
 
 2.使用模型LSTM
 X：Time
 Y：category

In [ ]:
# convert an array of values into a dataset matrix
def create_dataset(dataset, look_back=1):
    dataX, dataY = [], []
    for i in range(len(dataset)-look_back-1):
        a = dataset[i:(i+look_back), 0]
        dataX.append(a)
        dataY.append(dataset[i + look_back, 0])
    return numpy.array(dataX), numpy.array(dataY)

# fix random seed for reproducibility
numpy.random.seed(7)